src/blocks/form/edit.js

```diff
@@ -279,17 +273,22 @@ class FormEdit extends Component {
 	setTemplate( layout ) {
 		const { setAttributes } = this.props;
 		let submitButtonText;
+		let clientID = this.props.clientId;
 		map( TEMPLATE_OPTIONS, ( elem ) => {
 			if ( isEqual( elem.template, layout ) ) {
+				// // Update the child block's attributes
 				submitButtonText = elem.submitButtonText;
 				if ( Array.isArray( submitButtonText ) ) {
 					submitButtonText = submitButtonText.join( '' );
 				}
+				setTimeout( function() {
+					let childBlocks = wp.data.select( 'core/editor' ).getBlocksByClientId( clientID )[0].innerBlocks;
+					wp.data.dispatch('core/editor').updateBlockAttributes( childBlocks[ childBlocks.length - 1 ].clientId, { submitButtonText: submitButtonText } );
+				}, 100 );
```

Comment:
We should be able to detect when changes are made to the template using `componentDidUpdate` to then modify inner blocks. We shouldn't wait an arbitrary time when we know exactly when something has updated.